pkg/manifest/ostree_deployment.go

```diff
@@ -318,7 +305,7 @@ func (p *OSTreeDeployment) serialize() osbuild.Pipeline {
 		pipeline.AddStage(grpStage)
 	}
 
-	if p.ignition {
+	if p.IgnitionPlatform != "" {
```

Comment:
I suspect a lot of this is specific to IoT/Edge/Automotive

One thing that is interesting is that the `NewIgnitionStage` `Network:` stuff where `systemd.firstboot=off` and `systemd.condition-first-boot=true` get set is kind of an abuse of something that isn't really public API. It's meant to be an internal implementation detail.

Reply:
Again, I'll defer to @runcom.

Reply:
So, I think we can’t really say that’s not public api as it has been used in the past by coreos too (where we’ve learned about it) and even if undocumented it acts as an api to me 🤷🏻‍♂️ anyhow, I agree it’s a huge ack and it’s meant as a workaround that we need to solve (I’ll prioritize investigation and fix soon)